Console/Command/ReplicaDisableVirtualCommand.php

```diff
@@ -0,0 +1,178 @@
+    {
+        return 'Disable virtual replicas for all product sorting attributes and revert to standard replicas';
+    }
+
+    protected function getStoreArgumentDescription(): string
+    {
+        return 'ID(s) for store(s) to disable virtual replicas (optional), if not specified disable virtual replicas for all stores';
+    }
+
+    protected function getAdditionalDefinition(): array
+    {
+        return [];
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $this->output = $output;
+        $this->input = $input;
+        $this->setAreaCode();
+
+        $storeIds = $this->getStoreIds($input);
+
+        $output->writeln($this->decorateOperationAnnouncementMessage('Disabling virtual replicas for {{target}}', $storeIds));
+
+        $okMsg = 'Configure virtual replicas by attribute under: Stores > Configuration > Algolia Search > InstantSearch Results Page > Sorting';
+        if (!$this->confirmOperation($okMsg)) {
+            return CLI::RETURN_SUCCESS;
+        }
+
+        $this->disableVirtualReplicas($storeIds);
+
+        return Cli::RETURN_SUCCESS;
+    }
+
+    /**
+     * @param int[] $storeIds
+     * @return void
+     */
+    protected function disableVirtualReplicas(array $storeIds = []): void
+    {
+        $updates = [];
+        if (count($storeIds)) {
+            foreach ($storeIds as $storeId) {
+                if ($this->disableVirtualReplicasForStore($storeId)) {
+                    $updates[] = $storeId;
+                }
+            }
+            if ($updates) {
+                $this->scopeConfig->reinit();
+                foreach ($updates as $storeId) {
+                    $this->syncReplicasForStore($storeId);
+                }
+            }
+        } else {
+            $this->disableVirtualReplicasForAllStores();
+        }
+    }
+
+    protected function disableVirtualReplicasForStore(int $storeId): bool
+    {
+        $storeName = $this->storeNameFetcher->getStoreName($storeId);
+        $isStoreScoped = false;
+
+        if ($this->configChecker->isSettingAppliedForScopeAndCode(
+            ConfigHelper::LEGACY_USE_VIRTUAL_REPLICA_ENABLED,
+            ScopeInterface::SCOPE_STORES,
+            $storeId)
+        ) {
+            $isStoreScoped = true;
+            $this->removeLegacyVirtualReplicaConfig(ScopeInterface::SCOPE_STORES, $storeId);
+        }
+
+        if ($this->configChecker->isSettingAppliedForScopeAndCode(
+            ConfigHelper::SORTING_INDICES,
+            ScopeInterface::SCOPE_STORES,
+            $storeId)
+        ) {
+            $isStoreScoped = true;
+            $this->disableVirtualReplicaSortConfig(ScopeInterface::SCOPE_STORES, $storeId);
+        }
+
+        if (!$isStoreScoped) {
+            $this->output->writeln("<info>Virtual replicas are not configured at the store level for $storeName. You will need to re-run this command for all stores.</info>");
```

Comment:
Don't we want to allow to disable virtual replicas on one particular store even if there's no value set at store level ?

Reply:
This is a good question. In this scenario because the virtual replica is managed at a higher scope there's nothing to update because the store's virtual replica is either managed at the parent website scope or the root default scope.

If we wanted to silently handle this we would need to create an entirely new configuration and save it "on the fly" rather than edit the existing configuration which is how this command is intended to run. Not impossible but might have unintended side effects and go beyond the scope of the intended function. 

I think it's probably too much of an edge case to worry about too much. I see this CLI used in two contexts:

1. Facilitating upgrades from the old configuration and
1. Troubleshooting replica issues when limits are exceeded. 

In the first event, if they have to upgrade and are running into problems with the upgrade then they are best served by running the feature for all stores anyway. 

In the second event, the store level config is a convenience feature but not essential. They can always perform more fine grained configurations within the Magento admin.